### ISSUE TITLE
Error compile qtile

### ISSUE THREAD

**Issue by cr1ogen** [qtile/qtile#4774]
### Issue description

try to compile qtile git version give me this error
```
pip install --config-setting backend=wayland . --break-system-packages
DEPRECATION: Loading egg at /usr/local/lib/python3.11/dist-packages/pywlroots-0.16.6-py3.11-linux-x86_64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
Processing /home/cr1ogen/Git/qtile
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: cffi>=1.1.0 in /usr/lib/python3/dist-packages (from qtile==0.25.1.dev8+gb6d4a898) (1.16.0)
Requirement already satisfied: cairocffi>=1.6.0 in /usr/lib/python3/dist-packages (from cairocffi[xcb]>=1.6.0->qtile==0.25.1.dev8+gb6d4a898) (1.6.1)
Requirement already satisfied: xcffib>=1.4.0 in /usr/local/lib/python3.11/dist-packages (from qtile==0.25.1.dev8+gb6d4a898) (1.5.0)
Requirement already satisfied: pycparser in /usr/lib/python3/dist-packages (from cffi>=1.1.0->qtile==0.25.1.dev8+gb6d4a898) (2.22)
Building wheels for collected packages: qtile
  Building wheel for qtile (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for qtile (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [274 lines of output]
      Installing wlroots v0.16.7 requires wlroots v0.16.x, found v0.17.1
      Built with support for XWayland helpers.
      running bdist_wheel
      running build
      running build_py
      copying libqtile/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/command_client.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/command_graph.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/command_interface.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/command_object.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/configurable.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/group.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/images.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/ipc.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/lazy.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/log_utils.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/notify.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/pango_ffi.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/pangocffi.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/popup.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/sh.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/bar.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/config.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/dgroups.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/hook.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/utils.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/confreader.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/scratchpad.py -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/backend/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend
      copying libqtile/command/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/command
      copying libqtile/command/base.py -> build/lib.linux-x86_64-cpython-311/libqtile/command
      copying libqtile/command/client.py -> build/lib.linux-x86_64-cpython-311/libqtile/command
      copying libqtile/command/graph.py -> build/lib.linux-x86_64-cpython-311/libqtile/command
      copying libqtile/command/interface.py -> build/lib.linux-x86_64-cpython-311/libqtile/command
      copying libqtile/core/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/core
      copying libqtile/core/lifecycle.py -> build/lib.linux-x86_64-cpython-311/libqtile/core
      copying libqtile/core/loop.py -> build/lib.linux-x86_64-cpython-311/libqtile/core
      copying libqtile/core/state.py -> build/lib.linux-x86_64-cpython-311/libqtile/core
      copying libqtile/core/manager.py -> build/lib.linux-x86_64-cpython-311/libqtile/core
      copying libqtile/extension/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/extension
      copying libqtile/extension/base.py -> build/lib.linux-x86_64-cpython-311/libqtile/extension
      copying libqtile/extension/command_set.py -> build/lib.linux-x86_64-cpython-311/libqtile/extension
      copying libqtile/extension/dmenu.py -> build/lib.linux-x86_64-cpython-311/libqtile/extension
      copying libqtile/extension/window_list.py -> build/lib.linux-x86_64-cpython-311/libqtile/extension
      copying libqtile/interactive/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/interactive
      copying libqtile/interactive/iqshell_install.py -> build/lib.linux-x86_64-cpython-311/libqtile/interactive
      copying libqtile/interactive/iqshell_kernel.py -> build/lib.linux-x86_64-cpython-311/libqtile/interactive
      copying libqtile/layout/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/base.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/bsp.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/columns.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/floating.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/matrix.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/max.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/ratiotile.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/slice.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/spiral.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/stack.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/tile.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/verticaltile.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/zoomy.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/tree.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/xmonad.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/layout/screensplit.py -> build/lib.linux-x86_64-cpython-311/libqtile/layout
      copying libqtile/resources/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/resources
      copying libqtile/resources/sleep.py -> build/lib.linux-x86_64-cpython-311/libqtile/resources
      copying libqtile/resources/default_config.py -> build/lib.linux-x86_64-cpython-311/libqtile/resources
      copying libqtile/scripts/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/migrate.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/run_cmd.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/shell.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/start.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/top.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/udev.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/check.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/main.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/scripts/cmd_obj.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts
      copying libqtile/widget/canto.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/caps_num_lock_indicator.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/check_updates.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/chord.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/clipboard.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/cmus.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/config_error.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/countdown.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/cpu.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/crashme.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/crypto_ticker.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/currentlayout.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/currentscreen.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/df.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/generic_poll_text.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/gmail_checker.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/graph.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/idlerpg.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/image.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/import_error.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/khal_calendar.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/load.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/maildir.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/moc.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/mpd2widget.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/net.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/notify.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/nvidia_sensors.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/open_weather.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/prompt.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/pulse_volume.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/quick_exit.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/screensplit.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/sensors.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/sep.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/she.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/spacer.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/stock_ticker.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/systray.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/textbox.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/thermal_zone.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/volume.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/wallpaper.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/window_count.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/windowname.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/wttr.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/imapwidget.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/bluetooth.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/groupbox.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/keyboardkbdd.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/pomodoro.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/mpris2widget.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/widgetbox.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/statusnotifier.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/windowtabs.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/base.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/clock.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/tasklist.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/wlan.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/battery.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/keyboardlayout.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/launchbar.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/memory.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/backlight.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/widget/do_not_disturb.py -> build/lib.linux-x86_64-cpython-311/libqtile/widget
      copying libqtile/backend/base/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/base
      copying libqtile/backend/base/window.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/base
      copying libqtile/backend/base/drawer.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/base
      copying libqtile/backend/base/core.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/base
      copying libqtile/backend/wayland/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/drawer.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/layer.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/output.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/xdgwindow.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/window.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/xwindow.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/wlrq.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/core.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/wayland/inputs.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland
      copying libqtile/backend/x11/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/x11
      copying libqtile/backend/x11/drawer.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/x11
      copying libqtile/backend/x11/xcbq.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/x11
      copying libqtile/backend/x11/xcursors.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/x11
      copying libqtile/backend/x11/xcursors_ffi.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/x11
      copying libqtile/backend/x11/xkeysyms.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/x11
      copying libqtile/backend/x11/window.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/x11
      copying libqtile/backend/x11/core.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/x11
      copying libqtile/backend/wayland/cffi/build.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland/cffi
      copying libqtile/backend/wayland/cffi/cairo_buffer.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland/cffi
      copying libqtile/backend/wayland/cffi/libinput.py -> build/lib.linux-x86_64-cpython-311/libqtile/backend/wayland/cffi
      copying libqtile/resources/status_notifier/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/resources/status_notifier
      copying libqtile/resources/status_notifier/statusnotifieritem.py -> build/lib.linux-x86_64-cpython-311/libqtile/resources/status_notifier
      copying libqtile/scripts/migrations/__init__.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/_base.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/change_bitcoin.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/change_bluetooth_args.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/change_keychord_args.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/change_stockticker_args.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/change_widgetbox_args.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/module_renames.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/remove_cmd_prefix.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/rename_check_updates_widget.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/rename_hook.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/rename_threaded_poll_text.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/rename_tile_master.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/update_monad_args.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/update_togroup_args.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      copying libqtile/scripts/migrations/match_list_regex.py -> build/lib.linux-x86_64-cpython-311/libqtile/scripts/migrations
      running egg_info
      writing qtile.egg-info/PKG-INFO
      writing dependency_links to qtile.egg-info/dependency_links.txt
      writing entry points to qtile.egg-info/entry_points.txt
      writing requirements to qtile.egg-info/requires.txt
      writing top-level names to qtile.egg-info/top_level.txt
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files found matching 'libqtile/_ffi*.py'
      warning: no previously-included files found matching 'libqtile/backend/x11/_ffi*.py'
      warning: no previously-included files found matching 'libqtile/backend/wayland/_ffi*.*'
      warning: no previously-included files found matching 'libqtile/backend/wayland/_libinput*.*'
      no previously-included directories found matching 'rpm'
      warning: no previously-included files matching '__pycache__' found under directory '*'
      warning: no previously-included files matching '*.py[co]' found under directory '*'
      adding license file 'LICENSE'
      writing manifest file 'qtile.egg-info/SOURCES.txt'
      copying libqtile/py.typed -> build/lib.linux-x86_64-cpython-311/libqtile
      copying libqtile/resources/battery-icons/battery-caution-charging.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-caution.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-empty.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-full-charged.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-full-charging.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-full.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-good-charging.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-good.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-low-charging.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-low.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-missing.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/layout-icons/layout-bsp.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-columns.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-floating.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-matrix.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-max.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-monadtall.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-monadthreecol.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-monadwide.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-ratiotile.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-screensplit.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-slice.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-spiral.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-stack.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-tile.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-treetab.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-unknown.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-verticaltile.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-wmii.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-zoomy.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/battery-icons/battery-caution-charging.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-caution.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-empty.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-full-charged.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-full-charging.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-full.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-good-charging.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-good.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-low-charging.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-low.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/battery-icons/battery-missing.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/battery-icons
      copying libqtile/resources/layout-icons/layout-bsp.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-columns.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-floating.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-matrix.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-max.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-monadtall.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-monadthreecol.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-monadwide.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-ratiotile.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-screensplit.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-slice.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-spiral.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-stack.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-tile.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-treetab.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-unknown.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-verticaltile.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-wmii.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/layout-icons/layout-zoomy.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/layout-icons
      copying libqtile/resources/status_notifier/fallback_icon.png -> build/lib.linux-x86_64-cpython-311/libqtile/resources/status_notifier
      running build_ext
      generating cffi module 'build/temp.linux-x86_64-cpython-311/libqtile.backend.wayland._ffi.c'
      already up-to-date
      building 'libqtile.backend.wayland._ffi' extension
      x86_64-linux-gnu-gcc -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -Werror=implicit-function-declaration -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fPIC -DWLR_USE_UNSTABLE -I/usr/include/pixman-1 -I/usr/include/libdrm -I/tmp/pip-build-env-qv7zhzu7/normal/local/lib/python3.11/dist-packages/wlroots/include -I/usr/include/python3.11 -c build/temp.linux-x86_64-cpython-311/libqtile.backend.wayland._ffi.c -o build/temp.linux-x86_64-cpython-311/build/temp.linux-x86_64-cpython-311/libqtile.backend.wayland._ffi.o
      build/temp.linux-x86_64-cpython-311/libqtile.backend.wayland._ffi.c:588:10: fatal error: wlr/types/wlr_idle.h: No existe el fichero o el directorio
        588 | #include <wlr/types/wlr_idle.h>
            |          ^~~~~~~~~~~~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for qtile
Failed to build qtile
ERROR: Could not build wheels for qtile, which is required to install pyproject.toml-based projects
```

### Version

0.25.1.dev6+gf41c86b6

### Backend

Wayland (experimental)

### Required

- [X] I have searched past issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by elParaguayo** [comment]
You're building against wlroots 0.17 but we don't support that yet. 

Make sure 0.16 is installed and then export the environmental variables shown here: https://github.com/qtile/qtile/issues/4604#issuecomment-1868457613

**Comment by cr1ogen** [comment]
Ok, I'm going to try that, but until the last update it was compiling fine. Thank you

**Comment by cr1ogen** [comment]
Where is the PKGBUILD file in Qtile repo? I don't find it

**Comment by cr1ogen** [comment]
Anyway my Qtile works perfectly, i will wait more time

**Comment by elParaguayo** [comment]
The PKGBUILD isn't in the repo. That thread was about people building from the AUR but the issue is exactly the same.

This issue will be fixed once we merge support for 0.17.

**Comment by uncomfyhalomacro** [comment]
i seem to get the same issue btw. please reopen. 0.26.0 says that it supports wlroots 0.17.0.
but sources https://files.pythonhosted.org/packages/f0/a2/d9ded7b28884d54338d97659133bceacf76b34b27c7f2d897ec99cc173ab/qtile-0.26.0.tar.gz seems not
